commands/slash/clean.js

```diff
@@ -0,0 +1,46 @@
+const SlashCommand = require("../../lib/SlashCommand");
+
+const command = new SlashCommand()
+    .setName("clean")
+    .setDescription("Cleans the last 100 bot messages from channel.")
+    .addIntegerOption((option) =>
+        option
+            .setName("number")
+            .setDescription("Number of messages to delete.")
+            .setRequired(false)
+    )
+    .setRun(async (client, interaction, options) => {
+
+        await interaction.deferReply();
+        let number = interaction.options.getInteger("number");
+        number = number? ++number : 100;
+
+
+        interaction.channel.messages.fetch({
+            limit: number
+        }).then((messages) => {
+            const botMessages = [];
+            messages.filter(m => m.author.id === client.user.id).forEach(msg =>  botMessages.push(msg))
+
+            botMessages.shift();
+            interaction.channel.bulkDelete(botMessages, true)
+                .then(async deletedMessages => {
+                    //Filtering out messages that did not get deleted.
+                    messages = messages.filter(msg => {
+                        !deletedMessages.some(deletedMsg => deletedMsg == msg);
+                    });
+                    if (messages.size > 0) {
+                        client.log(`Deleting [${messages.size}] messages older than 14 days.`)
+                        messages.deleteAll();
```

Comment:
I need someone to test deletion of old bot messages older than 14 days. I couldn't test for this edge case.

Reply:
I'll test it it later, but like why a clean command tho it's it a music bot.. I have tons of my own commands top tho here is the list of commands that I made extra in discord music bot:

Bugreport Feedback 8ball
Wordle Minesweeper Eval
Execute Configinfo Banner Instagram Setstatus
And weather and they are all tested and work 100% hit me up on discord if you're interested in any of those files- Mex#1222

Reply:
> I need someone to test deletion of old bot messages older than 14 days

I don't think you can delete messages older than 2 weeks, it's discord limitation.